src/Compilers/CSharp/Test/Semantic/Semantics/TargetTypedDefaultTests.cs

```diff
@@ -47,6 +70,12 @@ static void Main()
             var comp = CreateCompilationWithMscorlib(source, parseOptions: TestOptions.ExperimentalParseOptions, options: TestOptions.DebugExe);
             comp.VerifyDiagnostics();
             CompileAndVerify(comp, expectedOutput: "0");
+
+            var tree = comp.SyntaxTrees.First();
+            var model = comp.GetSemanticModel(tree);
+            var def = tree.GetCompilationUnitRoot().DescendantNodes().OfType<DefaultLiteralSyntax>().Single();
+
+            Assert.Equal(0, model.GetConstantValue(def).Value);
```

Comment:
>model.GetConstantValue(def).Value [](start = 28, length = 33)

What type is returned by SemanticModel for the "def" node here? That type and the type of the constant value should agree.